### PR TITLE
Update hxCodec Libraries

### PR DESCRIPTION
**Pull request by ghost**
## Features
- Video Sprites (includes a Chart Editor event to trigger one)
- MacOS Support
- Helper Class for Videos

## Goal
Update the hxCodec Libraries from the base engine and provide helper functions for video sprites and lua callbacks to control such sprites

Solves https://github.com/ShadowMario/FNF-PsychEngine/issues/5401

## Progress
around 50% i'd say, pausing is still broken, normal videos haven't been tested but they likely work fine since no major changes were made to playback from what I've seen on hxCodec's source, the focus here is the video sprite event and the sprites themselves

### CODE REVIEW

**Review by DetectiveBaldi — APPROVED** [review]
good pr 
hopefully this gets merged whenever its done

**Comment by BeastlyGabi** [comment]
linux is broken thx to my laziness hold on

File: source/handlers/PsychVideo.hx [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+import flixel.FlxSprite;
+import flixel.FlxCamera;
+import flixel.FlxG;
+import openfl.Assets as OpenFlAssets;
+#if sys
+import sys.FileSystem;
+#end
+#if VIDEOS_ALLOWED
+import VideoHandler;
```
Comment (system32unknown) [inline source/handlers/PsychVideo.hx:11]: put hxcodec behind `VideoHandler`

**Review by UmbratheUmbreon — APPROVED** [review]

File: source/handlers/PsychVideo.hx (hunk 1 above)
Comment (BeastlyGabi) [reply source/handlers/PsychVideo.hx:11]: image
can't be done

**Review by DM-kun — APPROVED** [review]

File: Project.xml [hunk 2]
```diff
@@ -47,7 +47,7 @@
 	<define name="MODS_ALLOWED" if="desktop" />
 	<define name="LUA_ALLOWED" if="desktop" />
 	<define name="ACHIEVEMENTS_ALLOWED" />
-	<define name="VIDEOS_ALLOWED" if="web || windows" unless="32bits"/>
+	<define name="VIDEOS_ALLOWED" if="desktop || web" unless="32bits"/>
```
Comment (saturn-volv) [inline Project.xml:50]: hxCodec also supports android now

File: source/handlers/PsychVideo.hx (hunk 1 above)
Comment (saturn-volv) [reply source/handlers/PsychVideo.hx:11]: it's `hxcodec.VideoHandler`
image

**Review by Kn1ghtNight — APPROVED** [review]
such a good pr
now we gotta wait for it to be merged

File: Project.xml (hunk 2 above)
Comment (BeastlyGabi) [reply Project.xml:50]: psych doesn't natively support android anyways so no need

**Comment by teotm** [comment]
Linux would not compile hxCodec unless you use the git version of the repo. As for now, there aren't any major changes between 2.6.1 and git, so you can use the git version without being worried that a completely new error will occur.

**Comment by BeastlyGabi** [comment]
gonna close this until [hxCodec 3.0](https://github.com/EliteMasterEric/hxCodec) is released

**Comment by BeastlyGabi** [comment]
but in the meantime https://github.com/ShadowMario/FNF-PsychEngine/pull/11821

**Pull request closed without merging**